src/s1ap/handlers/s1ap_encoder.c

```diff
@@ -1267,3 +1267,124 @@ int s1ap_mme_encode_handover_mme_status_transfer(
     *length = enc_ret;
     return enc_ret;
 }
+
+int s1ap_mme_encode_handover_prep_failure(
+  struct handover_preparation_failure_Q_msg *s1apPDU,
+  uint8_t **buffer,
+  uint32_t *length)
+{
+    S1AP_PDU_t pdu = { (S1AP_PDU_PR_NOTHING) };
```

Comment:
we are doing memset below. In that case initializing pdu here may not be required.